### ISSUE TITLE
PathVerifier fails on whitespace filename under Linux and .NET Core

### ISSUE THREAD

**Issue by AnakinRaW** [TestableIO/System.IO.Abstractions#1070]
**Describe the bug**
When using `FileSystem` and `MockFileSystem` I noticed there is different behavior when creating a new `IFileInfo` using only whitespaces. 
The "error" trails down to the PathVerifier that the `MockFileSystem` uses

**To Reproduce**
```cs

// Both throw in Linux and windows
var mfs = new MockFileSystem();
mfs.FileInfo.New("   "); // ASCII space
mfs.FileInfo.New("\u00A0"); // Unicode char that's treated as whitespace

// Both pass under Linux
mfs.FileInfo.New("./   ") // Explicit relative path 
mfs.FileInfo.New("/   ") // Explicit absolute path 

// Passes in Linux 
var fs = new FileSystem();
fs.FileInfo.New("   "); // ASCII space

// Passes in Linux and Windows (.NET Core); throws on NET4.8
fs.FileInfo.New("\u00A0"); // Unicode char that's treated as whitespace
```

**Expected behavior**
Both calls throw an exception when they should not.
For Linux `   ` (whitespace) is a valid file name
For Windows and Linux `\u00A0` is a valid file name. However under .NET Framework it also fails

**Additional context**
In `PathInternal.OS.cs` .NET has a `bool IsEffectivelyEmpty(string? path)` method which behaves differently to linux and windows, just to cover those cases. 

Currently `PathVerifier` uses the `string.Trim()` method, which causes the two explained scenarios
- valid unicode chars get trimmed (`\u00A0`)
- a whitespace-only relative path is considered to be illegal when it actually is not under linux.

**Comment by fgreinacher** [comment]
Thanks for reporting! Happy to accept a PR to fix this 🙇

**Comment by bearpro** [comment]
Maybe just remove `.Trim() == string.Empty` check? I do not see anything similar in dotnet/runtime. Unix seems to interpret spaces literlly as provided. Windows with NTFS seems to trim ending spaces. 

The line with `.Trim()` was added in https://github.com/TestableIO/System.IO.Abstractions/commit/bce1401a68be0eda5af1c9122d721b6a67841473#diff-de56bbf9f4705d511bb1e432ef634a4656b9dd4cd46dff4338c085cd6ceca72cR27, and seem like just an assumption.

**Comment by vbreuss** [comment]
No, the check already existed previously (see [here](https://github.com/TestableIO/System.IO.Abstractions/blob/ee1fd19b39b69cf448cb179eeeb1cde78e691574/TestingHelpers/MockFile.cs#L273) and was only moved in this commit.

Originally this code was introduced with [#cdbf565](https://github.com/TestableIO/System.IO.Abstractions/pull/40/commits/cdbf5653c060e2c14856440cf385077507780a87):
> MockFile.Move handles now invalid filenames (blank, empty, null) with correct exceptions.

Maybe this check should only run under Windows?

I would be open for a pull request, @bearpro...
